source/slang/slang-reflection-api.cpp

```diff
@@ -1587,8 +1587,13 @@ namespace Slang
                     //
                     return;
                 }
-                else if(resourceKindCount == 1)
+                else
                 {
+                    // `resourceKindCount` should be 1 in most cases.
+                    // However if this field is a buffer of existential type,
+                    // the resourceInfo array will contain additional ExistentialTypeParam
+                    // and ExistentialObjectParam entries. These entries doesn't affect the
+                    // logic here, so we only need to care about the first entry.
```

Comment:
Should we do something like scanning through the list of resource kinds until we find one that is *not* the existential object/type case?

I worry that a change somewhere else that results in the resource kinds being listed in a different order could break this logic.